pkg/core/xds/metadata.go

```diff
@@ -117,6 +120,10 @@ func (m *DataplaneMetadata) GetDynamicMetadata(key string) string {
 	return m.DynamicMetadata[key]
 }
 
+func (m *DataplaneMetadata) GetVersion() *mesh_proto.Version {
```

Comment:
optional: I don't know how much do we rely on that, but the common pattern for implementing getters for DataplaneMetadata is checking `if m == nil` at the first line.